pkg/function/client/client.go

```diff
@@ -113,6 +113,45 @@ func (c *client) MapFn(ctx context.Context, datum *functionpb.DatumRequest) ([]*
 	return mappedDatumList.GetElements(), nil
 }
 
+// MapStreamFn applies a function to each datum element and returns a stream.
+func (c *client) MapStreamFn(ctx context.Context, datum *functionpb.DatumRequest) (<-chan functionpb.DatumResponse, <-chan error) {
+	resultCh := make(chan functionpb.DatumResponse)
+	errCh := make(chan error)
+	stream, err := c.grpcClt.MapStreamFn(ctx, datum)
+	if err != nil {
+		errCh <- fmt.Errorf("failed to execute c.grpcClt.MapStreamFn(): %w", err)
+		close(resultCh)
+		close(errCh)
+		return resultCh, errCh
+	}
+
+	go func() {
+		defer close(resultCh)
+		defer close(errCh)
+		for {
+			select {
+			case <-ctx.Done():
+				errCh <- ctx.Err()
+				return
+			default:
+				resp, err := stream.Recv()
+				if err == io.EOF {
+					return
+				}
+				if err != nil {
+					errCh <- err
+					return
+				}
+				for _, d := range resp.GetElements() {
```

Comment:
do we really need an error channel? if we just return when we get an error.

Reply:
is it like an error per message? or an error per stream?

Reply:
nvm, I checked the numaflow pr, it's one error per stream. So basically we are redelivering the message if we encounter an error and since the error could be because of any one of the responses from flatmap we need a channel. With dedup in place we are good.